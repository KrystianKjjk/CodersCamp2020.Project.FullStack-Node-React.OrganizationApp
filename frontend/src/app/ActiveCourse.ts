import { CourseListElementModel } from "../components/CourseList/CourseListSlice";

export function setActiveCourse(course:CourseListElementModel){
localStorage.setItem("activeCourse", JSON.stringify(course));
}

export function getActiveCourse(): CourseListElementModel | null{
const activeCourseItem = localStorage.getItem("activeCourse");
if(!activeCourseItem){
    return null;
}
return JSON.parse(activeCourseItem);
}

