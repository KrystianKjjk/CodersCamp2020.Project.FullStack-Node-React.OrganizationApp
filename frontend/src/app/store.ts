import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit'
import editingReducer from '../Redux/EditingSlice'
import reusableTableReducer from '../components/ReusableTable/ReusableTableSlice'
import homePageReducer from '../pages/Common/HomePage/HomePageSlice'
import courseDetails from '../pages/Admin/Course/CourseDetailsSlice'
import courseList from '../pages/Admin/CourseList/CourseListSlice'
import SnackbarReducer from '../components/Snackbar/SnackbarSlice'

export const store = configureStore({
  reducer: {
    editingInfo: editingReducer,
    tables: reusableTableReducer,
    userData: homePageReducer,
    courseDetails,
    courseList,
    snackbarDetails: SnackbarReducer,
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>
