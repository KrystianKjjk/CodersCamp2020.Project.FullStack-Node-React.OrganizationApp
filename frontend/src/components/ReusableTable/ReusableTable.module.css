.container {
    width: 100%;
    min-width: 300px;
    height: 400px;
}
.table {
    background-color: grey;
}