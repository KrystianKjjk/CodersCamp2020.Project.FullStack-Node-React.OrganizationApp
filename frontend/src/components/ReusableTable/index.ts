export { default } from './ReusableTable'
export * from './ReusableTable'
