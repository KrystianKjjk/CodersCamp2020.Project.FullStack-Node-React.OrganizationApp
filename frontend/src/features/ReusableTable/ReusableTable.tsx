import React from 'react';
import styles from './ReusableTable.module.css';

export interface ReusableTableProps {

}

const ReusableTable: React.FC< ReusableTableProps > = props => {
  return (
    <div>
      ReusableTable
    </div>
  );
};

export default ReusableTable;