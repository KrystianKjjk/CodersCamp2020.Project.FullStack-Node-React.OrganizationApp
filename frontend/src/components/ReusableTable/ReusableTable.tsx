import React, { useEffect } from 'react';
import { DataGrid } from '@material-ui/data-grid';
import styles from './ReusableTable.module.css';
import { useAppDispatch, useAppSelector } from '../../app/hooks';
import { 
  initTable,
  selectTables
} from './ReusableTableSlice';

interface Column {
  field: string;
  headerName?: string;
  width?: number;
  type?: string;
}

export interface ReusableTableProps {
  name: string;
  columns?: Column[];
  getEndpoint?: string;
  xAuthToken?: string;
}

const ReusableTable: React.FC< ReusableTableProps > = ({name, columns = []}) => {
  const rows = [{id: 1, name: 'Name1'}, {id: 2, name: 'Name2'}];
  const tables = useAppSelector(selectTables);
  const dispatch = useAppDispatch();

  useEffect(() => {
    dispatch(initTable({ name }));
  }, []);

  return (
    <div className={styles.container} aria-label={'Table - ' + name}>
      {name}
      <DataGrid
        rows={rows}
        columns={columns}
        pageSize={5}
        checkboxSelection
      />
    </div>
  );
};

export default ReusableTable;