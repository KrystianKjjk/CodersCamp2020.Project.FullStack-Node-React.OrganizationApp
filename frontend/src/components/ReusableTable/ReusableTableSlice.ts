import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { AppThunk, RootState } from '../../app/store';

interface ReusableTableState {
  [name: string]: {
    rows: any[];
  }
}

const initialState: ReusableTableState = {
  
};

export const reusableTableSlice = createSlice({
  name: 'reusableTable',
  initialState,
  reducers: {
    initTable: (state, { payload }) => {
      state[payload.name] = {
        rows: [],
      }
    },
  },
});

export const { initTable } = reusableTableSlice.actions;

// export const incrementAsync = (amount: number): AppThunk => dispatch => {
//   setTimeout(() => {
//     dispatch(incrementByAmount(amount));
//   }, 1000);
// };

// if you want, add selectors here, change the one below, remember to register reducer in store.ts
export const selectTables = (state: RootState) => state.tables;

export default reusableTableSlice.reducer;
