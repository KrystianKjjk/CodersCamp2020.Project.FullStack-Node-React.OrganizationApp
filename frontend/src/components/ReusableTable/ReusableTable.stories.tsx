import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import ReusableTable , { ReusableTableProps } from './ReusableTable';

export default {
  title: 'ReusableTable component',
  component: ReusableTable,
} as Meta;

const Template: Story<ReusableTableProps> = (args) => <ReusableTable {...args} />;

export const SampleReusableTable = Template.bind({});
SampleReusableTable.args = {
  name: 'My Table',
  columns: [{field: 'id', width: 100}, {field: 'name', width: 100}],
};