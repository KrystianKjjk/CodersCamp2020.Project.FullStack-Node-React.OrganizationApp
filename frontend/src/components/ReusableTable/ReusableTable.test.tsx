import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import { store } from '../../app/store';
import ReusableTable from './ReusableTable';

const tableName = 'Users';

describe('ReusableTable', () => {
   it('renders without error', () => {
      render(
         <Provider store={store}>
            <ReusableTable name={tableName}/>
         </Provider>
      );
      const table = screen.getByLabelText(`Table - ${tableName}`);
      expect(table).toMatchSnapshot();
   });

   it('test table initialization', () => {
      const state = store.getState();
      expect(state.tables[tableName]).toEqual({
         rows: [],
       });
   })
});