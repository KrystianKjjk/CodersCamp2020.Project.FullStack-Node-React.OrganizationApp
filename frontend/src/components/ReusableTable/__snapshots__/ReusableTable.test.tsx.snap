// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReusableTable renders without error 1`] = `
<div
  aria-label="Table - Users"
  class="container"
>
  Users
  <div
    aria-colcount="0"
    aria-label="grid"
    aria-multiselectable="false"
    aria-rowcount="2"
    class="MuiDataGrid-root MuiDataGrid-root"
    role="grid"
    tabindex="0"
  >
    <div />
    <div
      class="MuiDataGrid-main"
    >
      <div
        class="MuiDataGrid-columnsContainer"
        style="min-height: 56px; max-height: 56px; line-height: 56px;"
      >
        <div
          aria-rowindex="1"
          class="MuiDataGrid-colCellWrapper "
          role="row"
          style="transform: translate3d(-0px, 0, 0);"
        />
      </div>
      <div
        style="overflow: visible; height: 0px; width: 0px;"
      >
        <div
          style="width: 0px; height: 0px;"
        >
          <div
            class="MuiDataGrid-window"
            style="top: 56px; overflow-y: auto;"
          >
            <div
              class="MuiDataGrid-dataContainer data-container"
            >
              <div
                class="MuiDataGrid-viewport"
                style="min-width: 0; max-width: 0; min-height: 0; max-height: 0;"
              >
                <div
                  class="rendering-zone"
                  style="max-height: 0; width: 0px; pointer-events: none; transform: translate3d(-0px, -0px, 0);"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="Mui-resizeTriggers"
      >
        <div
          class="expand-trigger"
        >
          <div
            style="width: 1px; height: 1px;"
          />
        </div>
        <div
          class="contract-trigger"
        />
      </div>
    </div>
    <div>
      <div
        class="MuiDataGrid-footer"
      >
        <div />
        <div
          class="MuiTablePagination-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
          >
            <div
              class="MuiTablePagination-spacer"
            />
            <p
              class="MuiTypography-root MuiTablePagination-caption makeStyles-caption-2 MuiTypography-body2 MuiTypography-colorInherit"
            >
              1-2 of 2
            </p>
            <div
              class="MuiTablePagination-actions"
            >
              <button
                aria-label="Previous page"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                title="Previous page"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </span>
              </button>
              <button
                aria-label="Next page"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                title="Next page"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
